tests/components/utility_meter/test_select.py

```diff
@@ -1,12 +1,81 @@
 """The tests for the utility_meter select platform."""
 
+import pytest
+
 from homeassistant.components.utility_meter.const import DOMAIN
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.setup import async_setup_component
 
 from tests.common import MockConfigEntry
 
 
+@pytest.mark.parametrize(
+    ("yaml_config", "config_entry_config"),
+    [
+        (
+            {
+                "utility_meter": {
+                    "energy_bill": {
```

Comment:
I think you should put this in two different tests instead. It doesn't add anything to combine it in the same with an if-then-else.

I think you should test with a slug which is different than the name so we see the entity id gets correct.
Then a second test with the entity pre-created in the entity registry so we see it's not changing the entity id when it already exist.

Reply:
I was stressing the point that regardless of the method (yaml or config entry) the results must be the same

Reply:
I think that's fair but since the test has the slug and the name to result to the same thing.
With this test we can't see where it's grabbing the information from.

Reply:
tests have been split